src/ml/neural_net/tf_compute_context.cpp

```diff
@@ -269,5 +268,56 @@ tf_model_backend::~tf_model_backend() {
   call_pybind_function([&]() { model_ = pybind11::object(); });
 }
 
+tf_image_augmenter::tf_image_augmenter(const options& opts)
+    : processed_image_augmenter(opts) {}
+
+image_augmenter::result tf_image_augmenter::prepare_images(
+    std::vector<labeled_image> source_batch) {
+  return processed_image_augmenter::prepare_images(source_batch);
+}
+
+image_augmenter::intermediate_result
+tf_image_augmenter::prepare_augmented_images(
+    image_augmenter::intermediate_labeled_image data_to_augment) {
+  // const size_t n = opts_.batch_size;
+  image_augmenter::intermediate_result image_annotations;
+
+  call_pybind_function([&]() {
+    // Pass the required data to tensorflow
+    pybind11::module tf_aug = pybind11::module::import(
+        "turicreate.toolkits.object_detector._tf_image_augmenter");
+
+    // Get augmented images and annotations from tensorflow
+    pybind11::object tf_augmentation = tf_aug.attr("DataAugmenter")(
+        data_to_augment.images, data_to_augment.annotations,
+        data_to_augment.predictions);
+    pybind11::object augmented_img =
+        tf_augmentation.attr("get_augmented_images")();
```

Comment:
Do we need a `DataAugmenter` class in Python? Can this just be a function? Is there any state maintained from one augmentation batch to the next?

Reply:
As per our offline discussion, it was because of the heterogeneous data that was being passed back from python. I will change it to a function now that return a pair of pybind11 buffers and vector of pybind11 buffers.